From 1013326c7caad9b32569da472fcafad73fd8527f Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Sat, 25 May 2024 00:38:44 +0200
Subject: [PATCH] Fix missing return values

* return false for error conditions
* fixes return-mismatch issues which are fatal by default with gcc-14

oe-service/src/luna_methods.c:812:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:1532:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:1929:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:2006:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:2057:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:2137:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
oe-service/src/luna_methods.c:2217:3: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Submitted [https://github.com/webOS-ports/preware/pull/49]

 oe-service/src/luna_methods.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/oe-service/src/luna_methods.c b/oe-service/src/luna_methods.c
index a245230..82b7cec 100644
--- a/oe-service/src/luna_methods.c
+++ b/oe-service/src/luna_methods.c
@@ -809,7 +809,7 @@ void *update_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
 
  error:
   LSErrorPrint(&lserror, stderr);
@@ -1529,7 +1529,7 @@ void *feed_download_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
@@ -1926,7 +1926,7 @@ void *appinstaller_install_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
@@ -2003,7 +2003,7 @@ void *opkg_install_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
@@ -2054,7 +2054,7 @@ void *remove_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
@@ -2134,7 +2134,7 @@ void *opkg_replace_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
@@ -2214,7 +2214,7 @@ void *appinstaller_replace_thread(void *arg) {
 
  end:
   LSMessageUnref(message);
-  return;
+  return NULL;
  error:
   LSErrorPrint(&lserror, stderr);
   LSErrorFree(&lserror);
